loops/lua/main.lua

```diff
@@ -0,0 +1,24 @@
+arg = { ... } -- Capture command-line arguments
+Input = arg[1] or ""
+if Input == "" then
+    print("Invalid input (Blank Input)")
+else
+    local number = tonumber(Input)
+    if number == nil then
+        print("Error: Input cannot be converted to a number")
+    else
+        -- Successful number conversion
+        local r = (math.random(number))
+        local a = {} -- new array
+        for i = 1, 10000 do
+            a[i] = 0
+        end
+        for i = 1, 10000 do
+            for j = 1, 10000 do
```

Comment:
This is missing a 0? See [https://github.com/bddicken/languages/blob/main/loops/c/code.c#L10C25-L10C31](https://github.com/bddicken/languages/blob/main/loops/c/code.c#L10C25-L10C31).

Reply:
I have added it , I think , sorry it was confusing and I mis added the number of 0's

Reply:
I haven't really checked it , checking it right now

Reply:
Okay so this has become a lot slower except the luajit part , maybe my pr wasn't so impressive after all

 bash run.sh
Benchmark 1: lua main_optimized.lua 40
1950039
1950040
1950032
1950003
1950033
1950037
1950004
1950027
  Time (mean ± σ):     18.502 s ±  0.036 s    [User: 18.255 s, System: 0.005 s]
  Range (min … max):   18.442 s … 18.532 s    5 runs


Benchmark 2: lua main.lua 40
1950003
1950023
1950022
1950028
1950022
1950007
1950014
1950039
  Time (mean ± σ):     26.830 s ±  0.184 s    [User: 26.515 s, System: 0.011 s]
  Range (min … max):   26.685 s … 27.152 s    5 runs

Benchmark 3: luajit main_luajit.lua 40
1950032
1950038
1950025
1950002
1950008
1950001
1950004
1950012
  Time (mean ± σ):      1.312 s ±  0.019 s    [User: 1.296 s, System: 0.001 s]
  Range (min … max):    1.283 s …  1.335 s    5 runs